### DISCUSSION TITLE
Is it possible to enable pattern telemetry connections duplicates?

### DISCUSSION THREAD

**Discussion by argargreg** [nasa/fprime#1881] [Q&A] [answered]
Dear all,

We would like to add another custom TlmChan in parallel to the already-existing TlmChan component, such that telemetry goes to both components. The PcktSend ports will go to different byte stream model drivers. 

When implementing this topology, the build process issues an error that there are pattern telemetry connections duplicates, and aborts. Is it a limitation, or is this an error we can ignore to allow the build process to carry on?

Otherwise, is the best way forward to create a new component that has two separate PcktSend ports?

Thank you in advance.

Kind regards

**Comment by LeStarch** [comment, marked as answer]
Typically you'd put a com splitter before them to get multiple outputs, then wire one to each downstream component.

**Reply by argargreg** [reply]
Thanks for the quick reply. Creating another component came next in terms of solutions but a splitter actually seems straightforward. I'm curious about my more general question about whether it is possible without, but I might as well make a splitter now (especially given that "typically" hints at the fact it isn't possible) and mark this as closed to make way for more important matters

**Reply by argargreg** [reply]
Nevermind that, found the Svc.ComSplitter component - thanks

**Reply by argargreg** [reply]
For others that might be coming here wondering the same thing, here is the working implementation.

fpp:
```
@ A component for splitting a Tlm port into two paths
  passive component TlmSplitter {

    @ Tlm input port
    sync input port TlmRecv: Fw.Tlm

    @ Tlm output port A
    output port TlmOutA: Fw.Tlm

    @ Tlm output port B
    output port TlmOutB: Fw.Tlm

  }
```

cpp
```
void TlmSplitter ::
    TlmRecv_handler(
        const NATIVE_INT_TYPE portNum,
        FwChanIdType id,
        Fw::Time &timeTag,
        Fw::TlmBuffer &val
    )
  {
    //Passing over by reference => need to make a copy first
    Fw::Time timeTagCopy = timeTag;
    Fw::TlmBuffer valCopy = val;

    if( isConnected_TlmOutA_OutputPort(0) ) {
      TlmOutA_out(0, id, timeTagCopy, valCopy);
    }
    
    if(isConnected_TlmOutB_OutputPort(0) ) {
      TlmOutB_out(0, id, timeTagCopy, valCopy);
    }
  }
```

instances.fpp
`instance tlmSplitter: TlmSplitter base id 0x4C00`

topology.fpp
```
telemetry connections instance tlmSplitter
connections TelemSplitter {
      tlmSplitter.TlmOutA -> chanTlm.TlmRecv
    }
```

Thanks.
